### PR TITLE
Zoom independent distance matrix for fog

### PR DESCRIPTION
**Pull request by arindam1993**
- Calculate a terrain-independent zoom based on distance of camera from plane at z=0
- Add `cameraMatrix` which is the position of the camera in this space
- Refactor posMatrix to projMatrix to separate out tile projection and positioning logic.

TODO:
[ ] Make `cameraMatrix` account for translate properties

### CODE REVIEW

File: src/shaders/terrain_raster.vertex.glsl [hunk 1]
```diff
@@ -7,6 +7,7 @@ attribute vec2 a_texture_pos;
 varying vec2 v_pos0;
 
 #ifdef FOG
+uniform mat4 u_cam_matrix;
```
Comment (karimnaaji) [inline src/shaders/terrain_raster.vertex.glsl:10]: Do you think this should be directly put in the _prelude like the other fog specific uniforms or is that the only place we'd use this matrix? https://github.com/mapbox/mapbox-gl-js/blob/fog-implementation/src/shaders/_prelude_fog.fragment.glsl#L3_L6

Comment (arindam1993) [reply src/shaders/terrain_raster.vertex.glsl:10]: Yep that makes sense, the varying `v_depth` and any fog related declaration should as well the, right?

Comment (karimnaaji) [reply src/shaders/terrain_raster.vertex.glsl:10]: I think we could yes, I haven't put it there and it's duplicated but we can clean that to put in the prelude, and may wrap that transform function similarly to how fog is [applied](https://github.com/mapbox/mapbox-gl-js/blob/fog-implementation/src/shaders/_prelude_fog.fragment.glsl#L17) (so changing it globally is easier)

**Comment by pepe-invest-git** [comment]
MIL-MUN 0:2

**Pull request merged**